storops/unity/resource/lun.py

```diff
@@ -220,12 +220,16 @@ def attach_to(self, host, access_mask=HostLUNAccessEnum.PRODUCTION):
         return resp
 
     def detach_from(self, host):
-        if self.host_access is None or not host:
+        if self.host_access is None:
             return None
 
-        new_access = [{'host': item.host,
-                       'accessMask': item.access_mask} for item
-                      in self.host_access if host.id != item.host.id]
+        if host is None:
+            # Detach the lun from all hosts if `host` is None
```

Comment:
Better to make a log entry as well for easier triage.

Reply:
Sure.